exporters/prometheus/exporter_test.go

```diff
@@ -0,0 +1,132 @@
+	}{
+		{
+			name:         "counter",
+			expectedFile: "testdata/counter.txt",
+			recordMetrics: func(ctx context.Context, meter otelmetric.Meter) {
+				labels := []attribute.KeyValue{
+					attribute.Key("A").String("B"),
+					attribute.Key("C").String("D"),
+				}
+				counter, err := meter.SyncFloat64().Counter("foo", instrument.WithDescription("a simple counter"))
+				require.NoError(t, err)
+				counter.Add(ctx, 5, labels...)
+				counter.Add(ctx, 10.3, labels...)
+				counter.Add(ctx, 9, labels...)
+			},
+		},
+		{
+			name:         "gauge",
+			expectedFile: "testdata/gauge.txt",
+			recordMetrics: func(ctx context.Context, meter otelmetric.Meter) {
+				labels := []attribute.KeyValue{
+					attribute.Key("A").String("B"),
+					attribute.Key("C").String("D"),
+				}
+				gauge, err := meter.SyncFloat64().UpDownCounter("bar", instrument.WithDescription("a fun little gauge"))
+				require.NoError(t, err)
+				gauge.Add(ctx, 100, labels...)
+				gauge.Add(ctx, -25, labels...)
+			},
+		},
+		{
+			name:         "histogram",
+			expectedFile: "testdata/histogram.txt",
+			recordMetrics: func(ctx context.Context, meter otelmetric.Meter) {
+				labels := []attribute.KeyValue{
+					attribute.Key("A").String("B"),
+					attribute.Key("C").String("D"),
+				}
+				histogram, err := meter.SyncFloat64().Histogram("baz", instrument.WithDescription("a very nice histogram"))
+				require.NoError(t, err)
+				histogram.Record(ctx, 23, labels...)
+				histogram.Record(ctx, 7, labels...)
+				histogram.Record(ctx, 101, labels...)
+				histogram.Record(ctx, 105, labels...)
+			},
+		},
+		{
+			name:         "sanitized attributes to labels",
+			expectedFile: "testdata/sanitized_labels.txt",
+			recordMetrics: func(ctx context.Context, meter otelmetric.Meter) {
+				labels := []attribute.KeyValue{
+					// exact match, value should be overwritten
+					attribute.Key("A.B").String("X"),
+					attribute.Key("A.B").String("Q"),
+
+					// unintended match due to sanitization, values should be concatenated
+					attribute.Key("C.D").String("Y"),
+					attribute.Key("C/D").String("Z"),
+				}
+				counter, err := meter.SyncFloat64().Counter("foo", instrument.WithDescription("a sanitary counter"))
+				require.NoError(t, err)
+				counter.Add(ctx, 5, labels...)
+				counter.Add(ctx, 10.3, labels...)
+				counter.Add(ctx, 9, labels...)
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ctx := context.Background()
+
+			exporter := NewExporter()
+			provider := metric.NewMeterProvider(metric.WithReader(exporter))
+			meter := provider.Meter("testmeter")
+
+			registry := prometheus.NewRegistry()
+			err := registry.Register(exporter.Collector)
+			require.NoError(t, err)
+
+			tc.recordMetrics(ctx, meter)
+
+			file, err := os.Open(tc.expectedFile)
+			require.NoError(t, err)
+
+			err = testutil.GatherAndCompare(registry, file)
+			require.NoError(t, err)
+
+			err = file.Close()
+			require.NoError(t, err)
```

Comment:
The required `NoError` after `GatherAndCompare` could potentially leave the filehandle open.

```suggestion
			require.NoError(t, err)
			t.Cleanup(func() { require.NoError(t, file.Close()) })

			err = testutil.GatherAndCompare(registry, file)
			require.NoError(t, err)
```